### ISSUE TITLE
Wrong term result when contractions are used since 14.2.0

### ISSUE THREAD

**Issue by Systemcluster** [spencermountain/compromise#925]
Since `14.2.0`, contractions in the analyzed text result in wrong terms results.

Minimal reproduction case:

```js
import compromise from "compromise/three"

const string = `
Test
plane's plane's
`

const usedWords = new Map()
const data = compromise(string)
for (const term of data.terms().json()) {
    console.log(term)
    const word = term.terms[0].text
    if (word !== '') usedWords.set(word, (usedWords.get(word) ?? 0) + 1)
}

console.log(usedWords)
```

The expected output is `Map(2) { 'Test' => 1, "plane's" => 2 }`, which matches the output with `14.1.1`.
The actual output with `14.2.0` is `Map(1) { 'Test' => 2 }`.

**Comment by spencermountain** [comment]
hey Chris, yeah this one is sneaky because it's trying to resolve the ambiguous case of 
`'the plane's cool'`
vs
`the plane's captain`

not sure what it should do with "plane's plane's", but it's trying its best.

let me know if i've misunderstood the issue
cheers

**Comment by Systemcluster** [comment]
One issue is that it's a somewhat impactful behavior change in a minor version upgrade (it worked properly until `14.1.2`), and the other (arguably bigger) issue is that it affects preceding words - In the above example, `Test` will appear *twice* in the `terms` result, while it appears only once in the text. 

Take this other example:

```js
const string = `
Test
***
plane's plane's plane's plane's`
```

It will result in this output containing both `Test` and `***` twice, and no `plane's` :

```
{
  text: 'Test',
  terms: [
    {
      text: 'Test',
      pre: '\n',
      post: '\n',
      tags: [Array],
      normal: 'test',
      index: [Array],
      id: 'test|04Y00000I',
      chunk: 'Verb',
      dirty: true,
      switch: 'Noun|Verb'
    }
  ]
}
{
  text: '***',
  terms: [
    {
      text: '***\n',
      pre: '',
      post: '',
      tags: [Array],
      normal: '***',
      index: [Array],
      id: '***|04Y000013',
      confidence: 0.1,
      chunk: 'Noun'
    }
  ]
}
{
  text: 'Test',
  terms: [
    {
      text: 'Test',
      pre: '\n',
      post: '\n',
      tags: [Array],
      normal: 'test',
      index: [Array],
      id: 'test|04Y00000I',
      chunk: 'Verb',
      dirty: true,
      switch: 'Noun|Verb'
    }
  ]
}
{
  text: '***',
  terms: [
    {
      text: '***\n',
      pre: '',
      post: '',
      tags: [Array],
      normal: '***',
      index: [Array],
      id: '***|04Y000013',
      confidence: 0.1,
      chunk: 'Noun'
    }
  ]
}
Map(2) { 'Test' => 2, '***\n' => 2 }
```

**Comment by spencermountain** [comment]
yep, this looks like a bug
```js
let txt = `plane's plane's`
let doc = nlp(txt)
doc.terms().debug()
```
will check it out.

**Comment by spencermountain** [comment]
the indexing issues with multiple contractions should be fixed now in `14.2.1`
i don't know what the proper expansion of these examples should be, but surprised this was not found in the tests!
cheers

**Comment by Systemcluster** [comment]
Thanks for fixing both of these issues so fast! I can confirm everything is working properly now. 🙂
